### PR TITLE
JAMES-3775 Tests for null internalDate edge case (only when migration issue)

### PR DESCRIPTION
**Pull request by quantranhong1999**
This is only a rare case, and the current code is resilient enough IMO.

### CODE REVIEW

File: third-party/rspamd/src/main/java/org/apache/james/rspamd/task/GetMailboxMessagesService.java [hunk 1]
```diff
@@ -109,8 +109,8 @@ private Flux<MessageResult> getMailboxMessagesOfAUser(Username username, Mailbox
         return Mono.from(mailboxManager.getMailboxReactive(mailboxMetaData.getId(), mailboxSession))
             .map(Throwing.function(MessageManager::getMailboxEntity))
             .flatMapMany(Throwing.function(mailbox -> mapperFactory.getMessageMapper(mailboxSession).findInMailboxReactive(mailbox, MessageRange.all(), MessageMapper.FetchType.METADATA, UNLIMITED)))
-            .doOnNext(mailboxMessageMetaData -> context.incrementHamMessageCount())
             .filter(mailboxMessageMetaData -> afterDate.map(date -> mailboxMessageMetaData.getInternalDate().after(date)).orElse(true))
```
Comment (vttranlina) [inline third-party/rspamd/src/main/java/org/apache/james/rspamd/task/GetMailboxMessagesService.java:112]: Q:
why we don't check null for case `mailboxMessageMetaData.getInternalDate()` ?

Comment (quantranhong1999) [reply third-party/rspamd/src/main/java/org/apache/james/rspamd/task/GetMailboxMessagesService.java:112]: As I understand, we always write not null `internalDate` to our DB. This is only null when migration mistake.

Comment (quantranhong1999) [reply third-party/rspamd/src/main/java/org/apache/james/rspamd/task/GetMailboxMessagesService.java:112]: BTW I don't think we are accepting creating an object with null internal date (throw NPE like this) but predicate checking instead (throwing some IllegalArgumentException). Anyway, let's say RuntimeException generally and these tests are mainly proving what happened with a null internal date in the pre-prod Cassandra table.

**Comment by chibenwa** [comment]
Isn't it better to "JUST" run MessageDenormalizationMigration on the target envirnment?

Also, there is supposed to be fallback support in order to read the `internalDate` from `messagev3` when it is missing from either `imapUidTable` or `messageIdTable`. Such behavior is implemented in `CassandraMessageMapper` as well as `CassandraMessageIdMapper`.

This fix only scratches the surface of the issue and is unsatfying to me. We need to dig deeper.

**Comment by chibenwa** [comment]
SimpleMailboxMessage requires the internal date to be here. If it is missing -> error.

**Comment by chibenwa** [comment]
I would like to know `Why we have a message with a null date` in the first place...

**Comment by quantranhong1999** [comment]
> I would like to know Why we have a message with a null date in the first place...

https://github.com/apache/james-project/blob/master/upgrade-instructions.md#rework-message-denormalization

We spotted that we did finish that migration well for pre-prod during the Cassandra driver 4 upgradation.

**Comment by chibenwa** [comment]
Even if not applied it is not supposed to matter to be honnest.
